src/scheduler/fifo.cpp

```diff
@@ -1646,7 +1646,7 @@ run_update_resresv(status *policy, int pbs_sd, server_info *sinfo,
 					fflush(stdout);
 #endif /* localmod 031 */
 
-					pbsrc = run_job(get_svr_inst_fd(pbs_sd, resresv->job->svr_inst_id), rr, execvnode, sinfo->has_runjob_hook, err);
+					pbsrc = run_job(get_svr_inst_fd(pbs_sd, rr->job->svr_inst_id), rr, execvnode, sinfo->has_runjob_hook, err);
```

Comment:
Please have a shortcut in get_svr_inst_fd() if there is only 1 server.  No need to search by string if it is just one.

Reply:
Optimized get_svr_inst_fd() to return early in case of single server mode. Please check the latest commit. 

Please note that in case of single server we cannot return vfd itself. It is because it may not be the first fd of the server instances after the PR #2086.

Reply:
just to clarify, vfd and real fd will be the same in the case of a single server cluster. Returning the first fd also works.

Reply:
that's how it is right now, but aren't we planning to change it so that vfd is not the first real fd? To be able to distinguish between real and virtual fd in IFL ?

Reply:
I was talking about a single server cluster. For multi-server cluster vfd will be different than that of any instance fd.

Reply:
ah ok, makes sense